src/components/button/button.scss

```diff
@@ -659,12 +659,26 @@
 :host([scale="l"]:not([appearance="transparent"])) a.content--slotted {
   --calcite-button-padding-y: 11px;
 }
+//for 'l' solid buttons shrink the padding if icon is present to preserve the height
+:host([scale="l"][icon-start]:not([appearance="transparent"])) button.content--slotted,
```

Comment:
This is a pretty advanced selector. Should we move this logic into the component and add a class for it? Could always be a refactor issue if we don't want to do it now.

Reply:
I'm all up for doing it now.